### PR TITLE
Ordering process

### LINKED ISSUE #91: Backend: Ordering Process

### PR DESCRIPTION
**Pull request by Essiga**

### CODE REVIEW

File: src/SMAIAXBackend.API/Endpoints/Order/OrderSmartMeterConnectorEndpoint.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace SMAIAXBackend.API.Endpoints.Order;
+
+public static class OrderSmartMeterConnectorEndpoint
+{
+    public static async Task<Ok<Guid>> Handle(
+        IOrderService orderService,
+        IKeyGenerationService keyGenerationService)
+    {
+        var connectorSerialNumber = await orderService.OrderSmartMeterConnectorAsync();
```
Comment (1r-1na) [inline src/SMAIAXBackend.API/Endpoints/Order/OrderSmartMeterConnectorEndpoint.cs:15]: Wouldn't it be nicer to return a DTO? Now you don't really know the GUID of what is returned here.

File: src/SMAIAXBackend.API/Endpoints/Order/OrderEndpoints.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public static WebApplication MapOrderEndpoints(this WebApplication app)
+    {
+        const string contentType = "application/json";
+        var group = app.MapGroup("/api/orders")
+            .WithTags("Order")
+            .RequireAuthorization();
+
+        group.MapPost("/", OrderSmartMeterConnectorEndpoint.Handle)
+            .WithName("OrderSmartMeterConnector")
+            .Accepts<object>(contentType)
```
Comment (1r-1na) [inline src/SMAIAXBackend.API/Endpoints/Order/OrderEndpoints.cs]:
```suggestion
```
Remove this line. The request body is not used.
image

Without is better:
image

File: src/SMAIAXBackend.Domain/Model/Entities/SmartMeter.cs [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
+using SMAIAXBackend.Domain.Model.ValueObjects;
 using SMAIAXBackend.Domain.Model.ValueObjects.Ids;
 
 namespace SMAIAXBackend.Domain.Model.Entities;
 
 public sealed class SmartMeter : IEquatable<SmartMeter>
 {
     public SmartMeterId Id { get; } = null!;
+
+    public ConnectorSerialNumber ConnectorSerialNumber { get; private set; } = new ConnectorSerialNumber(Guid.Empty);
```
Comment (Smightym8) [inline src/SMAIAXBackend.Domain/Model/Entities/SmartMeter.cs:13]: Do we need to assign `new ConnectorSerialNumber(Guid.Empty);` here?

File: src/SMAIAXBackend.Domain/Model/ValueObjects/ConnectorSerialNumber.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿namespace SMAIAXBackend.Domain.Model.ValueObjects;
+
+public class ConnectorSerialNumber(Guid id) : ValueObject
+{
+    public Guid Id { get; } = id;
```
Comment (Smightym8) [inline src/SMAIAXBackend.Domain/Model/ValueObjects/ConnectorSerialNumber.cs]: Renamed `Id` to `SerialNumber`

**Review by Smightym8 — CHANGES_REQUESTED** [review]

File: src/SMAIAXBackend.Domain/Model/Entities/SmartMeter.cs (hunk 3 above)
Comment (Essiga) [reply src/SMAIAXBackend.Domain/Model/Entities/SmartMeter.cs:13]: I think so? Optimally we wouldn't need it but that would mean refactoring the current create smart meter process which will be done in a later step (create is changed to add as the create is already called when a smart meter connector is ordered)

**Review by Smightym8 — APPROVED** [review]

**Pull request merged**